src/inference/dev_api/remote_utils.hpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ie_ngraph_utils.hpp"
+#include "ie_remote_blob.hpp"
+#include "ie_remote_context.hpp"
+#include "openvino/runtime/iremote_context.hpp"
+
+namespace ov {
+namespace legacy_convert {
+
+INFERENCE_ENGINE_API_CPP(ov::SoPtr<ov::IRemoteContext>)
+convert_remote_context(const std::shared_ptr<InferenceEngine::RemoteContext>& context);
+INFERENCE_ENGINE_API_CPP(ie::Blob*) get_hardware_blob(ie::Blob* blob);
+
+class INFERENCE_ENGINE_API_CLASS(TensorHolder) {
+public:
+    TensorHolder(ov::SoPtr<ov::ITensor> tensor) : _tensor(tensor) {}
+
+    ov::SoPtr<ov::ITensor> get_tensor() const {
```

Comment:
nitpick: it can be `const ov::SoPtr<ov::ITensor>&`

Reply:
Fixed in https://github.com/openvinotoolkit/openvino/pull/18920/commits/2367f0a1959c79a43294f77c8860071463a73a9c